test/ZkMinterModTriggerV1.t.sol

```diff
@@ -483,3 +504,136 @@ contract MintAndTrigger is ZkMinterModTriggerV1Test {
     assertEq(token.balanceOf(_recipient), _amount);
   }
 }
+
+contract RecoverTokens is ZkMinterModTriggerV1Test {
+  address public ETH = address(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE);
+
+  function _grantMinterRole(address _minter) internal {
+    vm.prank(admin);
+    minterTrigger.grantRole(MINTER_ROLE, _minter);
+  }
+
+  function testFuzz_SendsTokensToRecoveryAddress(address _minter, uint256 _amount) public {
+    _grantMinterRole(_minter);
+    _amount = bound(_amount, 1, cappedMinter.CAP());
+    vm.prank(_minter);
+    minterTrigger.mint(address(minterTrigger), _amount);
+
+    vm.prank(admin);
+    minterTrigger.recoverTokens(address(token), _amount);
+
+    assertEq(token.balanceOf(address(minterTrigger)), 0);
+    assertEq(token.balanceOf(recoveryAddress), _amount);
+  }
+
+  function testFuzz_SendsETHToRecoveryAddress(uint256 _amount) public {
+    _amount = bound(_amount, 1, type(uint128).max);
+    vm.deal(address(minterTrigger), _amount);
+
+    vm.prank(admin);
+    minterTrigger.recoverTokens(ETH, _amount);
+
+    assertEq(address(minterTrigger).balance, 0);
+    assertEq(address(recoveryAddress).balance, _amount);
+  }
+
+  function testFuzz_EmitsTokensRecoveredEvent(address _minter, uint256 _amount) public {
+    _grantMinterRole(_minter);
+    _amount = bound(_amount, 1, cappedMinter.CAP());
+    vm.prank(_minter);
+    minterTrigger.mint(address(minterTrigger), _amount);
+
+    vm.prank(admin);
+    vm.expectEmit();
+    emit ZkMinterModTriggerV1.TokensRecovered(admin, address(token), _amount, recoveryAddress);
+    minterTrigger.recoverTokens(address(token), _amount);
+  }
+
+  function testFuzz_EmitsTokensRecoveredEventForETH(uint256 _amount) public {
+    _amount = bound(_amount, 1, type(uint128).max);
+    vm.deal(address(minterTrigger), _amount);
+
+    vm.prank(admin);
+    vm.expectEmit();
+    emit ZkMinterModTriggerV1.TokensRecovered(admin, ETH, _amount, recoveryAddress);
+    minterTrigger.recoverTokens(address(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE), _amount);
+  }
+
+  function testFuzz_RevertIf_NotAdmin(address _nonAdmin, address _minter, uint256 _amount) public {
+    _grantMinterRole(_minter);
+    vm.assume(_nonAdmin != admin);
+
+    _amount = bound(_amount, 1, cappedMinter.CAP());
+    vm.prank(_minter);
+    minterTrigger.mint(address(minterTrigger), _amount);
+
+    vm.prank(_nonAdmin);
+    vm.expectRevert(_formatAccessControlError(_nonAdmin, DEFAULT_ADMIN_ROLE));
+    minterTrigger.recoverTokens(address(token), _amount);
```

Comment:
Couldn't the token be fuzzed? Since it should revert before any token actions are take.

Reply:
Yes, you're right. Fuzzing now in https://github.com/zksync-association/zkminters/pull/34/commits/73cf84c33c0b3248481fac0ca14ef245df5c7b72